e2e_tests/tests/experiment/test_pytorch.py

```diff
@@ -280,3 +280,30 @@ def test_epoch_sync(num_workers: int, global_batch_size: int, dataset_len: int)
             assert exp.check_if_string_present_in_trial_logs(
                 t_id, f"rank {rank} finished batch {batch_idx} in epoch {epoch_idx}"
             )
+
+
+@pytest.mark.e2e_cpu
```

Comment:
can you write these as unit tests instead of e2e tests?

That will be far more performant, and more useful to developers.

Reply:
I thought about that for a while, and looked at the existing unit tests for inspiration, but given how little these callbacks do, the primary thing to check seems to be ensuring they fire where you expect them to.  I will look into doing this as a unit test somehow some more, but I wasn't able to see a way of implementing a useful unit test up to this point.

Reply:
Rewritten as unit tests, now that I understand our unit tests better.